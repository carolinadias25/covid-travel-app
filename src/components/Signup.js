/* eslint-disable no-console */
/* eslint-disable spaced-comment */
/* eslint-disable no-unused-vars */
/* eslint-disable quotes */
import React, { useState, useEffect } from "react";
import { Link, useHistory, Redirect } from "react-router-dom";
import firebase from "firebase";
import { FirebaseAuthConsumer } from "@react-firebase/auth";
import "../styles/Signup.css";

const Signup = () => {
  const [userName, setUserName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [passwordConfirm, setPasswordConfirm] = useState("");

  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);

  const history = useHistory();

  const signup = async () => {
    await firebase.auth().createUserWithEmailAndPassword(email, password);
  };

  useEffect(() => {
    const unsubscribe = firebase.auth().onAuthStateChanged((user) => {});
    return unsubscribe;
  }, []);

  const handleSubmitClick = async (event) => {
    event.preventDefault();

    if (password !== passwordConfirm) {
      setError("Passwords do not match");
    } else {
      try {
        setError("");
        setLoading(true);
        await signup();
        history.push("/");
      } catch (e) {
        setError("Failed to create account");
      } finally {
        setLoading(false);
      }
    }
  };

  return (
    <FirebaseAuthConsumer>
      {({ isSignedIn }) => {
        if (isSignedIn) {
          return <Redirect to="/" />;
        }
        return (
          <>
            {() => <p>Welcome! </p>}
            <h1 className="app-title">roamFree</h1>
            <div className="signup-section">
              <div className="signup-container">
                <form className="signup-form" onSubmit={handleSubmitClick}>
                  <div>
                    <input
                      id="name"
                      name="name"
                      type="text"
                      value={userName}
                      placeholder="Name"
                      onChange={(e) => {
                        setUserName(e.target.value);
                      }}
                    />
                  </div>
                  <div>
                    <input
                      id="email"
                      name="email"
                      type="text"
                      value={email}
                      placeholder="E-mail"
                      onChange={(e) => {
                        setEmail(e.target.value);
                      }}
                    />
                  </div>
                  <div>
                    <input
                      id="password"
                      name="password"
                      type="password"
                      value={password}
                      placeholder="Password"
                      onChange={(e) => {
                        setPassword(e.target.value);
                      }}
                    />
                  </div>
                  <div>
                    <input
                      id="passwordConfirm"
                      name="passwordConfirm"
                      type="password"
                      value={passwordConfirm}
                      placeholder="Confirm Password"
                      onChange={(e) => {
                        setPasswordConfirm(e.target.value);
                      }}
                    />
                  </div>
                  {error && <div>{error}</div>}
                  <button className="signup-button" type="submit">
                    SIGN UP
                  </button>
                  <div className="message-login-link">
                    Already a member?
                    <Link className="login-link" to="/login">
                      LOG IN
                    </Link>
                  </div>
                </form>
              </div>
            </div>
          </>
        );
      }}
    </FirebaseAuthConsumer>
  );
};

export default Signup;
