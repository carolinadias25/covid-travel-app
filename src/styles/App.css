/* .App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: black;
}
​
.App-link {
  color: black;
} */
​ .App {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
}
