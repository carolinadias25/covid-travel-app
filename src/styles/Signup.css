.signup-header {
  margin-top: 40px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: row;
      -ms-flex-align: row;
          align-items: row;
}

.error-message {
  font-family: "Nunito Sans", sans-serif;
  font-size: 15px;
  text-align: center;
  margin-bottom: 5px;
  color: rgba(0, 0, 0, 0.66);
  font-weight: bold;
}

.app-title {
  font-family: "Poppins", sans-serif;
  font-weight: 600;
  -ms-flex-item-align: end;
      align-self: flex-end;
  margin: 0;
  margin-left: 20px;
  font-size: 72px;
  line-height: 98px;
  color: #3ebbbf;
  text-shadow: 4px 4px 4px rgba(0, 0, 0, 0.07);
}

.signup-section {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 478px;
  max-width: 478px;
  height: 565px;
  left: 481px;
  top: 225px;
  vertical-align: middle;
  margin: auto;
  margin-top: 0;
  margin-bottom: 100px;
  background: #e5e5e5;
  -webkit-box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.25);
          box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.25);
  border-radius: 15px;
}

.signup-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 364px;
  height: 462px;
  left: 539px;
  top: 276px;
  vertical-align: middle;
  background: #ffffff;
  -webkit-box-shadow: 3px 4px 4px rgba(0, 0, 0, 0.15);
          box-shadow: 3px 4px 4px rgba(0, 0, 0, 0.15);
  border-radius: 15px;
}

.signup-form {
  width: 300px;
  height: 400px;
  margin: auto;
}

.message-login-link {
  width: 364px;
  height: 21px;
  left: 573px;
  top: 300px;
  font-family: "Poppins", sans-serif;
  font-style: normal;
  font-weight: 600;
  font-size: 17px;
  line-height: 23px;
  text-align: left;
  margin-left: 28px;
  margin-top: 10px;
  color: #ff8400;
}

.form__title {
  margin-bottom: 2rem;
  text-align: center;
}

.login-link {
  font-family: "Poppins", sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 17px;
  line-height: 23px;
  color: #3ebbbf;
  text-shadow: 4px 4px 4px rgba(0, 0, 0, 0.1);
  margin-left: 1em;
}

#name {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
          box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}

#name:focus {
  outline: none;
}

#email {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
          box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}

#email:focus {
  outline: none;
}

#password {
  width: 300px;
  height: 50px;
  background: rgba(229, 229, 229, 0.66);
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
          box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}

#password:focus {
  outline: none;
}

#passwordConfirm {
  width: 300px;
  height: 50px;
  background: rgba(229, 229, 229, 0.66);
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
          box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}

#passwordConfirm:focus {
  outline: none;
}

.signup-button {
  font-family: "Poppins", sans-serif;
  width: 300px;
  height: 55px;
  background: rgba(62, 187, 191, 0.66);
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
          box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: #ffffff;
  cursor: pointer;
  margin-bottom: 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.signup-button:hover {
  -webkit-transform: scale(1.05);
          transform: scale(1.05);
  cursor: pointer;
}
/*# sourceMappingURL=Signup.css.map */