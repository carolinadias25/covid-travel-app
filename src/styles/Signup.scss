.app-title {
  text-align: center;
  margin-top: 65px;
  font-size: 72px;
  line-height: 98px;
  font-weight: normal;
  color: #3ebbbf;
  text-shadow: 4px 4px 4px rgba(0, 0, 0, 0.07);
}
.signup-section {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 478px;
  max-width: 478px;
  height: 565px;
  left: 481px;
  top: 225px;
  vertical-align: middle;
  margin: auto;
  margin-top: 0;
  margin-bottom: 100px;

  background: #e5e5e5;
  box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.25);
  border-radius: 15px;
}

.signup-container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 364px;
  height: 462px;
  left: 539px;
  top: 276px;
  vertical-align: middle;

  background: #ffffff;
  box-shadow: 3px 4px 4px rgba(0, 0, 0, 0.15);
  border-radius: 15px;
}

.signup-form {
  width: 300px;
  height: 400px;
  margin: auto;
}

.message-login-link {
  width: 364px;
  height: 21px;
  left: 573px;
  top: 300px;

  font-family: Nunito Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 17px;
  line-height: 23px;
  text-align: left;
  margin-left: 40px;
  margin-top: 10px;

  color: rgba(238, 178, 22, 0.66);
}
.form__title {
  margin-bottom: 2rem;
  text-align: center;
}

.login-link {
  font-family: Nunito Sans;
  font-style: normal;
  font-weight: bold;
  font-size: 17px;
  line-height: 23px;
  color: #3ebbbf;
  text-shadow: 4px 4px 4px rgba(0, 0, 0, 0.1);
  margin-left: 1em;
}
#name {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}
#email {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}
#password {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}
#passwordConfirm {
  width: 300px;
  height: 50px;
  background: #fafafa;
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: rgba(0, 0, 0, 0.66);
  margin-bottom: 1.5em;
}
.signup-button {
  width: 300px;
  height: 55px;
  background: rgba(62, 187, 191, 0.66);
  mix-blend-mode: normal;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.05);
  border-radius: 10px;
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
  color: #ffffff;
  cursor: pointer;
  margin-bottom: 0;
}
