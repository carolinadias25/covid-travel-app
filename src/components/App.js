/* eslint-disable comma-dangle */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-console */
/* eslint-disable import/order */
/* eslint-disable quotes */

import React, { useState } from "react";
import PropTypes from "prop-types";
import "../styles/App.css";
import Login from "./Login";
import Signup from "./Signup";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { FirebaseAuthProvider } from "@react-firebase/auth";
import firebase from "firebase/app";
import "firebase/auth";
import NavBar from "./NavBar";
import CovidMap from "./CovidMap";
import HomePage from "./HomePage";
import Landing from "./LandingPage";
import PrivateRoute from "./PrivateRoute";
import CountryInfo from "./CountryInfo";
import ReactTooltip from "react-tooltip";

const config = {
  apiKey: "AIzaSyCksXWCl4y5g7yPE9tUD8Mv5PqktSVkADs",
  authDomain: "roamfree-dad3a.firebaseapp.com",
  projectId: "roamfree-dad3a",
  storageBucket: "roamfree-dad3a.appspot.com",
  messagingSenderId: "87295273544",
  appId: "1:87295273544:web:53aa041e58966f6a98c67d",
  measurementId: "G-H7C0CK2SNZ",
};

function App({ countries }) {
  const [content, setContent] = useState("");
  return (
    <FirebaseAuthProvider firebase={firebase} {...config}>
      <div className="App">
        <Router>
          <NavBar />
          <div>
            <Switch>
              <Route exact path="/" component={Landing} />
              <PrivateRoute path="/homepage">
                <HomePage />
              </PrivateRoute>
              <PrivateRoute path="/country-info/:country">
                <CountryInfo countries={countries} />
              </PrivateRoute>
              <PrivateRoute path="/covid-map">
                <CovidMap setTooltipContent={setContent} />
                <ReactTooltip>{content}</ReactTooltip>
              </PrivateRoute>
              <Route path="/login" component={Login} />
              <Route path="/signup" component={Signup} />
            </Switch>
          </div>
        </Router>
      </div>
    </FirebaseAuthProvider>
  );
}

App.propTypes = {
  countries: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      vaccinated: PropTypes.string.isRequired,
      unvaccinated: PropTypes.string.isRequired,
      quarantine: PropTypes.string.isRequired,
      masks: PropTypes.string.isRequired,
      restaurants: PropTypes.string.isRequired,
      bars: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default App;
