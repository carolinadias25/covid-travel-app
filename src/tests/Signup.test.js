/* eslint-disable */
import React from "react";
import { render } from "@testing-library/react";
import firebase from "firebase";
import Signup from "../components/Signup";
import App from "../components/App";

jest.mock("firebase", () => {
  return {
    auth: jest.fn(),
    initializeApp: jest.fn(),
    app: jest.fn(),
  };
});

describe("Signup", () => {
  it("renders correctly", () => {
    firebase.auth.mockReturnValueOnce({
      currentUser: { email: "example@gmail.com", uid: 1, emailVerified: true },
    });
    const { asFragment } = render(
      <App countries={[]}>
        <Signup />
      </App>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
