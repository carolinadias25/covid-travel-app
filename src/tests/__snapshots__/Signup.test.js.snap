// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <ul
        class="navbar-links"
      />
      <div
        class="navbar-right"
      >
        <img
          alt=""
          id="navbar-logo"
          src="appLogo.png"
        />
      </div>
    </div>
    <div>
      <div
        class="landing-page"
      >
        <div
          class="landing-message"
        >
          <div
            class="landing-header"
          >
            <h2>
              roamSafe
            </h2>
            <img
              alt=""
              id="landing-app-logo"
              src="appLogo.png"
            />
          </div>
          <div
            class="tag-line"
          >
            <h3>
              Travel internationally, covid secure ðŸŒŽ
            </h3>
            <div
              class="landing-links"
            >
              <div>
                <a
                  class="signup-link-landing"
                  href="/signup"
                >
                  SIGN UP
                </a>
              </div>
              <div>
                <a
                  class="login-link-landing"
                  href="/login"
                >
                  LOG IN
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
